src/rust/engine/src/nodes.rs

```diff
@@ -495,11 +550,10 @@ pub struct Snapshot(pub Key);
 
 impl Snapshot {
   async fn create(context: Context, path_globs: PreparedPathGlobs) -> NodeResult<store::Snapshot> {
-    // Recursively expand PathGlobs into PathStats.
-    // We rely on Context::expand tracking dependencies for scandirs,
+    // We rely on Context::expand_globs tracking dependencies for scandirs,
     // and store::Snapshot::from_path_stats tracking dependencies for file digests.
```

Comment:
Honestly, I don't really know what this comment means. I think it refers to Pantsd memoization?

Reply:
`Node`s depend on one another in the `Graph` so that when a file changes on disk, we can invalidate the appropriate nodes. Those two methods ends up creating `Node` instances (`DigestFile`, `ReadLink`, `Scandir`) under the hood which indicate which files they depend on.